### PR TITLE
Creator test

### PR DESCRIPTION
**Pull request by OksanaBisyk-TomTom**
dev
Added XMLCreator for our project.

Files to check:
XMLCreator.h
XMLCreator.cpp

## Code reviewers

- [x] @CharlesLyleLaRue 
- [x] @Globalika 
- [ ] @Hipst0man 

### Second Level Review

- [ ] @Stepbura

### CODE REVIEW

File: XMLCreator/XMLCreator.cpp [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+			}
+		}
+	}
+
+	std::ofstream file{ filename };
+	std::shared_ptr<XMLNode> current_node = m_root;
+	Symbols symbol;
+	size_t depth = 0;
+	WriteTreeToFile(current_node, depth, file, symbol);
+	return true;
```
Comment (CharlesLyleLaRue) [inline XMLCreator/XMLCreator.cpp:324]: You forgot to close your file.

File: XMLCreator/XMLCreator.cpp [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+		ChildAppend(parent_node, new_node);
+	}
+	parent_node = m_root->children[(int)Children::COMMUNICATION];
+	for (size_t i = 0; i < COMMUNICATION_TAGS_SIZE; i++)
+	{
+		CreateElement(new_node, communication_tags[i], communication_values[i]);
+		ChildAppend(parent_node, new_node);
+	}
+	parent_node = m_root->children[(int)Children::LOGGING];
+	for (size_t i = 0; i < LOGGING_TAGS_SIZE; i++)
```
Comment (CharlesLyleLaRue) [inline XMLCreator/XMLCreator.cpp]: Maybe there're a better way to create one 'for' loop for all this cases. They are almost the same.

**Review by CharlesLyleLaRue — CHANGES_REQUESTED** [review]
Close a file after done with writing.
And check if there are a possibility for refactoring some repeated 'for' loops.

**Review by Globalika — DISMISSED** [review]

File: XMLCreator/XMLCreator.cpp (hunk 1 above)
Comment (OksanaBisyk-TomTom) [reply XMLCreator/XMLCreator.cpp:324]: File closed.

File: XMLCreator/XMLCreator.cpp (hunk 2 above)
Comment (OksanaBisyk-TomTom) [reply XMLCreator/XMLCreator.cpp]: CreateStandartTree() method was optimized.

**Review by CharlesLyleLaRue — APPROVED** [review]
Looks good.

**Review by Globalika — APPROVED** [review]

**Review by Stepbura — APPROVED** [review]

**Pull request merged**